pkgs/by-name/xr/xrock/package.nix

```
@@ -0,0 +1,41 @@
+{
+  fetchFromGitHub,
+  lib,
+  libusb1,
+  pkg-config,
+  stdenv,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xrock";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "xboot";
+    repo = "xrock";
+    tag = "v${version}";
+    hash = "sha256-4b/Us0UybWL2XQOqwR36TAEUmbx3WOtKi6H9VYetweA=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ libusb1 ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp xrock $out/bin
+
+    runHook postInstall
+  '';
```

```suggestion
  postPatch = ''
    substituteInPlace Makefile \
      --replace-fail '/usr/local/bin/xrock' '${placehoulder "out"}/bin/xrock' \
      --replace-fail '/etc/udev/rules.d/99-xrock.rules' '${placeholder "out"}/etc/udev/rules.d/99-xrock.rules' \
      --replace-fail '/usr/share/licenses/xrock/LICENSE' '${placeholder "out"}/share/licenses/xrock/LICENSE'
  '';
```
This also takes care of the udev rules